src/main_thread/tracks_store/tracks_store.ts

```diff
@@ -492,6 +528,119 @@ export default class TracksStore extends EventEmitter<ITracksStoreEvents> {
+      // or without video because of the option was set to "continue".
+      log.warn(`TS: No playable ${bufferType}, continuing without ${bufferType}`);
+      this.trigger("noPlayableTrack", {
+        trackType: bufferType,
+        period: { id: period.id, start: period.start, end: period.end },
+      });
+      // The previous event trigger could have had side-effects, so we
+      // re-check if we're still mostly in the same state
+      if (this._isDisposed) {
+        return; // Someone disposed the `TracksStore` on the previous side-effect
+      }
+    } else if (firstPlayableAdaptation === undefined) {
+      const noRepErr = new MediaError(
+        "NO_PLAYABLE_REPRESENTATION",
+        `No ${bufferType} Representation can be played`,
+        { tracks: undefined },
+      );
+      this.trigger("noPlayableTrack", {
+        trackType: bufferType,
+        period: {
+          id: period.id,
+          start: period.start,
+          end: period.end,
+        },
+      });
+      // The previous event trigger could have had side-effects, so we
+      // re-check if we're still mostly in the same state
+      if (this._isDisposed) {
+        return; // Someone disposed the `TracksStore` on the previous side-effect
+      }
+      this.trigger("error", noRepErr);
+      this.dispose();
+      return;
+    }
+    let typeInfo = getPeriodItem(this._storedPeriodInfo, period.id)?.[bufferType];
+    if (isNullOrUndefined(typeInfo)) {
+      return;
+    }
+    const selectedAdaptation = getPeriodItem(this._storedPeriodInfo, period.id)?.[
+      bufferType
+    ]?.storedSettings;
+    if (selectedAdaptation === null) {
+      // The track type has been explicitly disabled, there is no need to select
+      // a new track as a fallback.
+      return;
+    }
+
+    const switchingMode =
+      bufferType === "audio" ? this._defaultAudioTrackSwitchingMode : "reload";
+    const storedSettings =
+      firstPlayableAdaptation !== undefined
+        ? {
+            adaptation: firstPlayableAdaptation,
+            switchingMode,
+            lockedRepresentations: new SharedReference<IRepresentationsChoice | null>(
+              null,
+            ),
+          }
+        : null;
+    typeInfo.storedSettings = storedSettings;
+
+    this.trigger("trackUpdate", {
+      period: toExposedPeriod(period),
+      trackType: bufferType,
+      reason: "no-playable-representation",
+    });
+
+    // The previous event trigger could have had side-effects, so we
+    // re-check if we're still mostly in the same state
+    if (this._isDisposed) {
+      return; // Someone disposed the `TracksStore` on the previous side-effect
+    }
+    typeInfo = getPeriodItem(this._storedPeriodInfo, period.id)?.[bufferType];
+    if (isNullOrUndefined(typeInfo) || typeInfo.storedSettings !== storedSettings) {
+      return;
+    }
+    typeInfo.dispatcher?.updateTrack(storedSettings);
```

Comment:
Aren't we in the risk of setting `null` for the video track when there's already no audio track and vice-versa here? Shouldn't we also check for the `NO_AUDIO_VIDEO_TRACKS` error?